### PR TITLE
Some improvements

### PR DESCRIPTION
**Pull request by osa1**

### CODE REVIEW

File: src/bytebuffer.inl [hunk 1]
```diff
@@ -14,15 +14,7 @@ static void bytebuffer_reserve(struct bytebuffer *b, int cap) {
...
-	char *newbuf = malloc(cap);
-	if (b->len > 0) {
-		// copy what was there, b->len > 0 assumes b->buf != null
-		memcpy(newbuf, b->buf, b->len);
-	}
-	if (b->buf) {
-		// in case there was an allocated buffer, free it
-		free(b->buf);
-	}
+	char *newbuf = realloc(b->buf, cap);
```
Comment (rofl0r) [inline src/bytebuffer.inl]: realloc() can fail. it will return NULL then, so the assignments below should be in a `if(newbuf) { ... }` block.

Comment (osa1) [inline src/bytebuffer.inl]: `malloc()` can fail too and the existing code does not check for error conditions. Error checking is left as it was before in this patch.

Comment (rofl0r) [inline src/bytebuffer.inl]: well, you were mentioning "improvements" ...

**Comment by nsf** [comment]
So, hm... I don't know. Ok, I'll merge it.

**Pull request merged**